src/solvers/dg_curved/dg_3d.jl

```diff
@@ -88,26 +89,65 @@ function calc_interface_flux!(u::AbstractArray{<:Any,5}, mesh::CurvedMesh{3},
     # Interfaces in x-direction (`orientation` = 1)
     calc_interface_flux!(elements.surface_flux_values,
                          elements.left_neighbors[1, element],
-                         element, 1, u, mesh, equations, dg, cache)
+                         element, 1, u, surface_flux, mesh, equations, dg, cache)
     
     # Interfaces in x-direction (`orientation` = 2)
     calc_interface_flux!(elements.surface_flux_values,
                          elements.left_neighbors[2, element],
-                         element, 2, u, mesh, equations, dg, cache)
+                         element, 2, u, surface_flux, mesh, equations, dg, cache)
 
     # Interfaces in x-direction (`orientation` = 3)
     calc_interface_flux!(elements.surface_flux_values,
                          elements.left_neighbors[3, element],
-                         element, 3, u, mesh, equations, dg, cache)
+                         element, 3, u, surface_flux, mesh, equations, dg, cache)
   end
 
   return nothing
 end
 
 
 @inline function calc_interface_flux!(surface_flux_values, left_element, right_element, orientation, u, 
-                              mesh::CurvedMesh{3}, equations, dg::DG, cache)
-  @unpack surface_flux = dg
+                                      surface_flux, mesh::CurvedMesh{3}, equations, dg::DG, cache)
+
+  right_direction = 2 * orientation
+  left_direction = right_direction - 1
+
+  for j in eachnode(dg), i in eachnode(dg)
+    if orientation == 1
+      u_ll = get_node_vars(u, equations, dg, nnodes(dg), i, j, left_element)
+      u_rr = get_node_vars(u, equations, dg, 1,          i, j, right_element)
+
+      # First contravariant vector Ja^1 as SVector
+      normal_vector = get_contravariant_vector(1, cache, 1, i, j, right_element)
```

Comment:
I just noticed this here, but it seems weird to have an accessor like `get_contravariant_vector` that accepts a `cache` instead of the array from which something is extracted (as we node for the nodal variables, coordinates etc.). Maybe rewrite `get_contravariant_vector` accordingly? This would also make it more versatile.

Reply:
Done